test/e2e/service.go

```diff
@@ -1055,21 +1049,14 @@ func testNotReachable(ip string, port int) {
 		Failf("got port==0 for non-reachability check", url)
 	}
 
-	for t := time.Now(); time.Since(t) < podStartTimeout; time.Sleep(5 * time.Second) {
-		resp, err = httpGetNoConnectionPool(url)
+	expectNoError(wait.Poll(5*time.Second, podStartTimeout, func() (bool, error) {
```

Comment:
Please add a logging line above here that says what we're trying to do (something along the lines of "ensuring %s is not reachable")